pkg/restore/conflict_test.go

```diff
@@ -0,0 +1,216 @@
+				{ "name": "default-token-abcde" }
+			]
+		}
+		`,
+	),
+
+	unstructuredOrDie(
+		`
+		{
+			"kind": "ServiceAccount",
+			"apiVersion": "v1",
+			"metadata": {
+				"namespace": "ns1",
+				"name": "default"
+			},
+			"secrets": [
+				{ "name": "default-token-xzy12" }
+			]
+		}
+		`,
+	),
+}
+
+func BenchmarkMergeServiceAccountUnstructuredBasic(b *testing.B) {
+	var desired *unstructured.Unstructured
+
+	for n := 0; n < b.N; n++ {
+		desired, _ = mergeServiceAccountUnstructured(basicCase[0], basicCase[1])
+	}
+
+	result = desired
+}
+
+func BenchmarkMergeServiceAccountBasic(b *testing.B) {
+	var desired *unstructured.Unstructured
+
+	for n := 0; n < b.N; n++ {
+		desired, _ = mergeServiceAccounts(basicCase[0], basicCase[1])
+	}
+
+	result = desired
+}
+
+var multipleSecrets = [...]*unstructured.Unstructured{
+	unstructuredOrDie(
+		`
+		{
+			"apiVersion": "v1",
+			"kind": "ServiceAccount",
+			"metadata": {
+				"namespace": "ns1",
+				"name": "default"
+			},
+			"secrets": [
+				{ "name": "default-token-abcde" },
+				{ "name": "my-secret" },
+				{ "name": "sekrit" }
+			]
+		}
+		`,
+	),
+
+	unstructuredOrDie(
+		`
+		{
+			"kind": "ServiceAccount",
+			"apiVersion": "v1",
+			"metadata": {
+				"namespace": "ns1",
+				"name": "default"
+			},
+			"secrets": [
+				{ "name": "default-token-xzy12" },
+				{ "name": "my-old-secret" },
+				{ "name": "secrete"}
+			]
+		}
+		`,
+	),
+}
+
+func BenchmarkMergeServiceAccountUnstructuredMultipleSecrets(b *testing.B) {
+	var desired *unstructured.Unstructured
+
+	for n := 0; n < b.N; n++ {
+		desired, _ = mergeServiceAccountUnstructured(multipleSecrets[0], multipleSecrets[1])
+	}
+
+	result = desired
+}
+
+func BenchmarkMergeServiceAccountMultipleSecrets(b *testing.B) {
+	var desired *unstructured.Unstructured
+
+	for n := 0; n < b.N; n++ {
+		desired, _ = mergeServiceAccounts(multipleSecrets[0], multipleSecrets[1])
+	}
+
+	result = desired
+}
+
+var labelsAndAnnotations = [...]*unstructured.Unstructured{
+	unstructuredOrDie(
+		`
+		{
+			"apiVersion": "v1",
+			"kind": "ServiceAccount",
+			"metadata": {
+				"namespace": "ns1",
+				"name": "default",
+				"labels": {
+					"l1": "v1",
+					"l2": "v2",
+					"l3": "v3"
+				},
+				"annotations": {
+					"a1": "v1",
+					"a2": "v2",
+					"a3": "v3",
+					"a4": "v4"
+				}
+			},
+			"secrets": [
+				{ "name": "default-token-abcde" }
+			]
+		}
+		`,
+	),
+
+	unstructuredOrDie(
+		`
+		{
+			"kind": "ServiceAccount",
+			"apiVersion": "v1",
+			"metadata": {
+				"namespace": "ns1",
+				"name": "default",
+				"labels": {
+					"l1": "v1",
+					"l2": "v2",
+					"l3": "v3",
+					"l4": "v4",
+					"l5": "v5"
+				},
+				"annotations": {
+					"a1": "v1",
+					"a2": "v2",
+					"a3": "v3",
+					"a4": "v4",
+					"a5": "v5",
+					"a6": "v6"
+				}
+			},
+			"secrets": [
+				{ "name": "default-token-xzy12" }
+			]
+		}
+		`,
+	),
+}
+
+func BenchmarkMergeServiceAccountUnstructuredLabelsAndAnnotations(b *testing.B) {
```

Comment:
Please describe what this is testing in more detail

Reply:
The current structure of these benchmark tests was definitely quick & dirty - would it be preferable to move them into sub-benchmarks to get better organization and naming?

Reply:
Yes, let's do sub-benchmarks please